tests/pipeline/nodes/model/mask_rcnnv1/test_mask_rcnn.py

```diff
@@ -0,0 +1,213 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from unittest import mock
+
+import cv2
+import numpy as np
+import numpy.testing as npt
+import pytest
+import yaml
+import torch
+
+from peekingduck.pipeline.nodes.base import WeightsDownloaderMixin
+from peekingduck.pipeline.utils.bbox.transforms import xyxy2xyxyn
+from peekingduck.pipeline.nodes.model.mask_rcnn import Node
+from tests.conftest import PKD_DIR, get_groundtruth
+
+GT_RESULTS = get_groundtruth(Path(__file__).resolve())
+NP_FILE = np.load(Path(__file__).resolve().parent / "mask_rcnn_gt_masks.npz")
+
+
+@pytest.fixture
+def mask_rcnn_config():
+    with open(PKD_DIR / "configs" / "model" / "mask_rcnn.yml") as infile:
+        node_config = yaml.safe_load(infile)
+    node_config["root"] = Path.cwd()
+    node_config["iou_threshold"] = 0.5
+    node_config["score_threshold"] = 0.5
+    node_config["mask_threshold"] = 0.5
+
+    return node_config
+
+
+@pytest.fixture(
+    params=[
+        {"key": "score_threshold", "value": -0.5},
+        {"key": "score_threshold", "value": 1.5},
+        {"key": "mask_threshold", "value": -0.5},
+        {"key": "mask_threshold", "value": 1.5},
+        {"key": "iou_threshold", "value": -0.5},
+        {"key": "iou_threshold", "value": 1.5},
+        {"key": "min_size", "value": 0},
+        {"key": "max_size", "value": 0},
+        {"key": "max_num_detections", "value": 0},
+    ],
+)
+def mask_rcnn_bad_config_value(request, mask_rcnn_config):
+    mask_rcnn_config[request.param["key"]] = request.param["value"]
+    return mask_rcnn_config
+
+
+@pytest.fixture(params=["r50-fpn"])
+def mask_rcnn_type(request, mask_rcnn_config):
+    mask_rcnn_config["model_type"] = request.param
+    return mask_rcnn_config
+
+
+@pytest.mark.mlmodel
+class TestMaskRCNN:
+    def test_no_human_image(self, no_human_image, mask_rcnn_type):
+        no_human_img = cv2.imread(no_human_image)
+        mask_rcnn_type["score_threshold"] = 0.9
+        mask_rcnn = Node(config=mask_rcnn_type)
+        output = mask_rcnn.run({"img": no_human_img})
+        expected_output = {
+            "bboxes": np.empty((0, 4), dtype=np.float32),
+            "bbox_labels": np.empty((0)),
+            "bbox_scores": np.empty((0), dtype=np.float32),
+            "masks": np.empty((0, 0, 0), dtype=np.uint8),
+        }
+        assert output.keys() == expected_output.keys()
+        npt.assert_equal(output["bboxes"], expected_output["bboxes"])
+        npt.assert_equal(output["bbox_labels"], expected_output["bbox_labels"])
+        npt.assert_equal(output["bbox_scores"], expected_output["bbox_scores"])
+        npt.assert_equal(output["masks"], expected_output["masks"])
+
+    def test_detect_human_bboxes(self, human_image, mask_rcnn_type):
```

Comment:
I'm failing this test when running on my linux machine. Snippet of the logs below:
```
    def test_detect_human_bboxes(self, human_image, mask_rcnn_type):
        human_img = cv2.imread(human_image)
        image_size = human_img.size
        mask_rcnn = Node(config=mask_rcnn_type)
        output = mask_rcnn.run({"img": human_img})
    
        assert "bboxes" in output
        assert output["bboxes"].size > 0
    
        assert "masks" in output
        assert output["masks"].size > 0
    
        model_type = mask_rcnn.config["model_type"]
        image_name = Path(human_image).stem
        expected = GT_RESULTS[model_type][image_name]
        expected_mask = NP_FILE[image_name]
>       npt.assert_allclose(output["bboxes"], expected["bboxes"], atol=1e-3)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0.001
E       
E       Mismatched elements: 1 / 88 (1.14%)
E       Max absolute difference: 0.0011032
E       Max relative difference: 0.0034594
E        x: array([[0.492317, 0.394699, 0.909877, 0.994834],
E              [0.180765, 0.224991, 0.530412, 0.997762],
E              [0.265676, 0.328424, 0.28124 , 0.375933],...
E        y: array([[0.4923, 0.3947, 0.91  , 0.995 ],
E              [0.1808, 0.2249, 0.5304, 0.9978],
E              [0.2657, 0.3284, 0.2812, 0.3759],...

tests/pipeline/nodes/model/mask_rcnnv1/test_mask_rcnn.py:105: AssertionError
```
Have you tried running the Post Merge workflow on your branch to see if the test passes on both platforms?

Reply:
I have modified the Ground truth and added new tests and ground truth for the maskrcnn r101 variant, and I have ran pytest specifically for `test_mask_rcnn.py` on both windows 11 and ubuntu 18 environment. It manages to pass both tests.